include/jni/functions.hpp

```diff
@@ -145,14 +145,18 @@ namespace jni
 
 
     template < class T >
-    T* NewLocalRef(JNIEnv& env, T* ref)
+    UniqueLocalRef<T> NewLocalRef(JNIEnv& env, T* t)
        {
-        return CheckJavaException(env, Wrap<T*>(env.NewLocalRef(Unwrap(ref))));
+        jobject* obj = Wrap<jobject*>(env.NewLocalRef(Unwrap(t)));
+        CheckJavaException(env);
+        if (t && !obj)
+            throw std::bad_alloc();
+        return UniqueLocalRef<T>(reinterpret_cast<T*>(obj), LocalRefDeleter(env));
        }
 
-    inline void DeleteLocalRef(JNIEnv& env, jobject* localRef)
+    inline void DeleteLocalRef(JNIEnv& env, UniqueLocalRef<jobject>&& ref)
        {
-        env.DeleteLocalRef(Unwrap(localRef));
+        env.DeleteLocalRef(Unwrap(ref.release()));
         CheckJavaException(env);
```

Comment:
I'm not sure why `DeleteLocalRef` calls `CheckJavaException` but neither `DeleteGlobalRef` nor `DeleteWeakGlobalRef` do.